### PR TITLE
snippets: Add SystemView RTT tracing snippet

### PR DESCRIPTION
**Pull request by MaureenHelm**
Adds a new snippet to enable tracing with SEGGER SystemView and RTT.

Signed-off-by: Maureen Helm <maureen.helm@analog.com>

### CODE REVIEW

File: snippets/rtt-tracing/README.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+.. code-block:: console
+
+   west build -S rtt-tracing [...]
+
+Overview
+********
+
+This snippet enables SEGGER SystemView support with the tracing subsystem and
+redirects the serial console output to SEGGER RTT.
```
Comment (henrikbrixandersen) [inline snippets/rtt-tracing/README.rst:14]:
```suggestion
redirects the console output to SEGGER RTT.
```

It's not a given that a board has a serial console.

File: snippets/rtt-tracing/rtt-tracing.conf [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_USE_SEGGER_RTT=y
+CONFIG_RTT_CONSOLE=y
```
Comment (henrikbrixandersen) [inline snippets/rtt-tracing/rtt-tracing.conf:5]: We already have a `rtt-console` snippet. Could the console redirection be dropped from this snippet?

File: snippets/rtt-tracing/rtt-tracing.conf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+CONFIG_DEBUG_OPTIMIZATIONS=n
+CONFIG_DEBUG_THREAD_INFO=y
+
+CONFIG_TRACING=y
+CONFIG_SEGGER_SYSTEMVIEW=y
+CONFIG_SEGGER_SYSTEMVIEW_BOOT_ENABLE=n
+CONFIG_SEGGER_SYSVIEW_POST_MORTEM_MODE=n
+
+CONFIG_IDLE_STACK_SIZE=2048
```
Comment (henrikbrixandersen) [inline snippets/rtt-tracing/rtt-tracing.conf:16]: Is this size portable?

**Comment by teburd** [comment]
Love the idea

File: snippets/rtt-tracing/rtt-tracing.conf (hunk 2 above)
Comment (MaureenHelm) [reply snippets/rtt-tracing/rtt-tracing.conf]: Initially I had trouble getting SystemView tracing to work without redirecting the console, but I took another look after your comment and it appears to work just fine. Pushed an update that drops the redirection from the snippet.

File: snippets/rtt-tracing/rtt-tracing.conf (hunk 3 above)
Comment (MaureenHelm) [reply snippets/rtt-tracing/rtt-tracing.conf]: Not really, no. This was an artifact of extracting `samples/subsys/tracing/prj_sysview.conf` into a snippet, but it seems to not be necessary. Removed.

**Review by henrikbrixandersen — DISMISSED** [review]

File: snippets/rtt-tracing/README.rst
Comment (ubieda) [inline snippets/rtt-tracing/README.rst:1]: We may want to update SystemView doc pointing at this snippet. Reference: https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/tracing/index.rst#segger-systemview-support

**Review by ubieda — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]

**Review by ubieda — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

File: samples/subsys/tracing/sample.yaml [hunk 4]
```diff
@@ -28,7 +28,7 @@ tests:
       - mimxrt1064_evk
     integration_platforms:
       - nrf52840dk/nrf52840
-    extra_args: CONF_FILE="prj_sysview.conf"
```
Comment (nashif) [inline samples/subsys/tracing/sample.yaml:31]: with removal of this file, it is not possible to build this sample with sysview with snippets similar to all others. I suggest adding this to the documentation and describe how to build this on the command line with snippets.

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by nashif — APPROVED** [review]

**Review by teburd — APPROVED** [review]
I love this snippet, thanks for making it.

**Review by ubieda — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**